File: Cpp/621_Task Scheduler.cpp
class Solution {
public:
    int leastInterval(vector<char>& tasks, int n) {
        map<char, int> frequencys;
        int max_frequency = 0;
        for(auto& task : tasks){
            max_frequency = max(max_frequency, ++frequencys[task]);
        }
        
        int remain = 0;
        for(auto& frequency: frequencys){
            if(frequency.second == max_frequency){
                ++remain;
            }
        }
        
        return max(static_cast<int>(tasks.size()), (n + 1) * (max_frequency - 1) + remain);
    }
};

File: Kotlin/621_Task_Scheduler.kt
class Solution {
    fun leastInterval(tasks: CharArray, n: Int): Int {
        val frequency = IntArray(26)
        var maxFrequency = 0
        
        for(task in tasks) {
            maxFrequency = maxOf(maxFrequency, ++frequency[task - 'A'])
        }

        val remain = frequency.count { it == maxFrequency }

        return maxOf(tasks.size, (n + 1) * (maxFrequency - 1) + remain)
    }
}